From 0a9da2a5c6a321567381dc5ca0ba37b959b05261 Mon Sep 17 00:00:00 2001
From: Peter Oliver <git@mavit.org.uk>
Date: Mon, 15 Jul 2024 16:19:10 +0100
Subject: [PATCH 3/4] Mark multiple mml-sec tests as unstable when built in an
 RPM

* test/lisp/gnus/mml-sec-tests.el: Mark unstable tests which have been
seen to intermittently fail at
https://koji.fedoraproject.org/koji/packageinfo?packageID=560
---
 test/lisp/gnus/mml-sec-tests.el | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/test/lisp/gnus/mml-sec-tests.el b/test/lisp/gnus/mml-sec-tests.el
index 7abd1bd65a3..a20cf84858c 100644
--- a/test/lisp/gnus/mml-sec-tests.el
+++ b/test/lisp/gnus/mml-sec-tests.el
@@ -210,6 +210,7 @@ mml-secure-test-key-fixture
 
 (ert-deftest mml-secure-key-checks ()
   "Test mml-secure-check-user-id and mml-secure-check-sub-key on sample keys."
+  :tags (if (getenv "RPM_BUILD_ROOT") '(:unstable))
   (skip-unless (test-conf))
   (mml-secure-test-fixture
    (lambda ()
@@ -278,6 +279,7 @@ mml-secure-key-checks
 
 (ert-deftest mml-secure-find-usable-keys-1 ()
   "Make sure that expired and disabled keys and revoked UIDs are not used."
+  :tags (if (getenv "RPM_BUILD_ROOT") '(:unstable))
   (skip-unless (test-conf))
   (mml-secure-test-fixture
    (lambda ()
@@ -369,6 +371,7 @@ mml-secure-find-usable-keys-2
 
 (ert-deftest mml-secure-select-preferred-keys-1 ()
   "If only one key exists for an e-mail address, it is the preferred one."
+  :tags (if (getenv "RPM_BUILD_ROOT") '(:unstable))
   (skip-unless (test-conf))
   (mml-secure-test-fixture
    (lambda ()
@@ -407,6 +410,7 @@ mml-secure-select-preferred-keys-2
 
 (ert-deftest mml-secure-select-preferred-keys-3 ()
   "Expired customized keys are removed if multiple keys are available."
+  :tags (if (getenv "RPM_BUILD_ROOT") '(:unstable))
   (skip-unless (test-conf))
   (mml-secure-test-fixture
    (lambda ()
@@ -581,6 +585,7 @@ mml-secure-test-en-decrypt-with-passphrase
 (ert-deftest mml-secure-en-decrypt-1 ()
   "Encrypt message; then decrypt and test for expected result.
 In this test, the single matching key is chosen automatically."
+  :tags (if (getenv "RPM_BUILD_ROOT") '(:unstable))
   (skip-unless (test-conf))
   (dolist (method (enc-standards) nil)
     ;; no-exp@example.org with single encryption key
@@ -625,6 +630,7 @@ mml-secure-en-decrypt-3
 (ert-deftest mml-secure-en-decrypt-4 ()
   "Encrypt message; then decrypt and test for expected result.
 In this test, encrypt-to-self variables are set to lists."
+  :tags (if (getenv "RPM_BUILD_ROOT") '(:unstable))
   (skip-unless (test-conf))
   ;; Send from sub@example.org, which has two keys; encrypt to both.
   (let ((mml-secure-openpgp-encrypt-to-self
-- 
2.46.2

